src/test/java/uk/gov/companieshouse/api/testdata/service/impl/AcspProfileServiceImplTest.java

```diff
@@ -83,6 +92,142 @@ void createAcspProfileWithDefaultValues() throws DataException {
+                captured.getAmlDetails().get(1).getSupervisoryBody());
+        assertEquals("Membership Details 2",
+                captured.getAmlDetails().get(1).getMembershipDetails());
+        assertEquals("Supervisory Body 3",
+                captured.getAmlDetails().get(2).getSupervisoryBody());
+        assertEquals("Membership Details 3",
+                captured.getAmlDetails().get(2).getMembershipDetails());
+    }
+
+    private AcspProfile getAcspProfile(AcspProfileSpec spec) {
+        AcspProfile savedProfile = createSavedProfile();
+
+        List<AmlDetails> amlDetailsList = new ArrayList<>();
+        for (AmlSpec amlSpec : spec.getAmlDetails()) {
+            AmlDetails amlDetails = new AmlDetails();
+            amlDetails.setSupervisoryBody(amlSpec.getSupervisoryBody());
+            amlDetails.setMembershipDetails(amlSpec.getMembershipDetails());
+            amlDetailsList.add(amlDetails);
+        }
+        savedProfile.setAmlDetails(amlDetailsList);
+        return savedProfile;
+    }
+
+    private AcspProfileSpec getAcspProfileSpec() {
+        AcspProfileSpec spec = new AcspProfileSpec();
+        spec.setStatus("active");
+        spec.setType("ltd");
+
+        // Creating multiple AML details entries
+        AmlSpec amlSpec1 = new AmlSpec();
+        amlSpec1.setSupervisoryBody("Supervisory Body 1");
+        amlSpec1.setMembershipDetails("Membership Details 1");
+
+        AmlSpec amlSpec2 = new AmlSpec();
+        amlSpec2.setSupervisoryBody("Supervisory Body 2");
+        amlSpec2.setMembershipDetails("Membership Details 2");
+
+        AmlSpec amlSpec3 = new AmlSpec();
+        amlSpec3.setSupervisoryBody("Supervisory Body 3");
+        amlSpec3.setMembershipDetails("Membership Details 3");
+
+        spec.setAmlDetails(List.of(amlSpec1, amlSpec2, amlSpec3));
+        return spec;
+    }
+
+    @Test
+    void createAcspProfileWithNullAmlDetails() throws DataException {
+        AcspProfileSpec spec = new AcspProfileSpec();
+        spec.setStatus("active");
+        spec.setType("ltd");
+        spec.setAmlDetails(null); // Setting AmlDetails as null
+
+        when(randomService.getString(8)).thenReturn("randomId");
+        AcspProfile savedProfile = createSavedProfile();
+
+        when(repository.save(any(AcspProfile.class))).thenReturn(savedProfile);
+
+        AcspProfileData result = service.create(spec);
+
+        assertNotNull(result);
+        assertEquals("randomId", result.getAcspNumber());
+
+        ArgumentCaptor<AcspProfile> captor = ArgumentCaptor.forClass(AcspProfile.class);
+        verify(repository).save(captor.capture());
+
+        AcspProfile captured = captor.getValue();
+        assertNotNull(captured);
+        assertEquals("randomId", captured.getId());
+        assertEquals("randomId", captured.getAcspNumber());
+        assertEquals(spec.getStatus(), captured.getStatus()); // Default value
+        assertEquals(spec.getType(), captured.getType()); // Default value
```

Comment:
These aren't the default values but the ones in the spec - please remove the comments